griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,270 @@
+                    Literal(
+                        "file_path",
+                        description="Name of the file to be created, which will be used to save content in."
+                    ): str,
+                    Literal(
+                        "content",
+                        description="Content to be saved in Google Doc."
+                    ): str
+                }
+            ),
+        }
+    )
+    def save_content_to_google_doc(self, params: dict) -> ErrorArtifact | InfoArtifact:
+        doc_creation_result = self.create_google_doc({"file_name": params["file_name"]})
+    
+        if isinstance(doc_creation_result, ErrorArtifact):
+            return ErrorArtifact(f"error creating Google Doc: {doc_creation_result.value}")
+    
+        document_id = doc_creation_result.value.split()[-1]
+        save_content_params = {"document_id": document_id, "content": params["content"]}
+    
+        try:
+            saved_document_id = self._save_to_doc(save_content_params)
+            return InfoArtifact(saved_document_id)
+        except Exception as e:
+            return ErrorArtifact(f"error saving content to Google Doc with Id {document_id}: {str(e)}")
+
+    @activity(
+        config={
+            "description": "Can be used to download multiple Google Docs from Google Drive based on their paths.",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_paths",
+                        description="Destination file paths on Google Drive in the POSIX format. "
+                                    "For example, 'foo/bar/baz.txt, foo/bar/baz2.txt'"
+                    ): [str]
+                }
+            ),
+        }
+    )
+    def download_google_docs(self, params: dict) -> ListArtifact | ErrorArtifact:
+        file_paths = params["file_paths"]
+        downloaded_files = []
+
+        for file_path in file_paths:
+            document_artifact = self._download_document(file_path)
+            if isinstance(document_artifact, ErrorArtifact):
+                return ErrorArtifact(f"error downloading Google Doc for path {file_path}: {document_artifact.value}")
+            downloaded_files.append(document_artifact)
+
+        return ListArtifact(downloaded_files)
+
+    def _download_document(self, file_path: str) -> BlobArtifact | ErrorArtifact:
+
+        try:
+            service = self._build_client(self.DRIVE_SCOPES, "v3", "drive")
+            file_id = self._path_to_file_id(service, file_path)
+            if file_id:
+                results = service.files().list(q=f"name='{file_path}'", spaces="drive", pageSize=10).execute()
+                items = results.get("files", [])
+                if items:
+                    file_id = items[0]["id"]
+                    request = service.files().export(fileId=file_id, mimeType="text/plain")
+                    response = request.execute()
+                    return BlobArtifact(response)
```

Comment:
We know that Google Docs documents are always text. Why not use `TextArtifact` here?

Reply:
Done!